src/pipelinedefinition.cpp

```diff
@@ -792,4 +828,56 @@ Status PipelineDefinition::getOutputsInfo(tensor_map_t& outputsInfo, const Model
     return StatusCode::OK;
 }
 
+Status PipelineDefinition::getCustomNodeMetadata(const NodeInfo& customNodeInfo, tensor_map_t& inputsInfo, metadata_fn callback) const {
+    struct CustomNodeTensorInfo* info = nullptr;
+    int infoLength = 0;
+    auto paramArray = createCustomNodeParamArray(customNodeInfo.parameters);  // TODO: not create it in every call? prepare it once?
```

Comment:
We don't change metadata in between PipelineDefinition reloads. Maybe just call it once after succesfull validation & then reset it during reload? We should add coment in appropiate place that we can do this since DL is not unloaded when removed from library manager (since library manager can start unloading before pipeline can reload).

Reply:
I would say that it depends on how much time it would take to have it prepared.

Reply:
There still is some trade off. If we want to prepare it once and use it outside watcher thread (in pipeline execution (runtime) validation) it needs to be thread safe. It would probably be fine if we used it in definition validation only.

Reply:
Done: https://jira.devtools.intel.com/browse/CVS-49029